tests/platform_tests/link_flap/test_cont_link_flap.py

```diff
@@ -0,0 +1,133 @@
+
+        Test steps:
+            1.) Flap all interfaces one by one in 1-3 iteration
+                to cause BGP Flaps.
+            2.) Flap all interfaces on peer (FanOutLeaf) one by one 1-3 iteration
+                to cause BGP Flaps.
+            3.) Watch for memory (show system-memory) ,orchagent CPU Utilization
+                and Redis_memory.
+
+        Pass Criteria: All routes must be re-learned with < 5% increase in Redis and 
+            ORCH agent CPU consumption below threshold after 3 mins after stopping flaps.
+        """
+        orch_cpu_threshold = request.config.getoption("--orch_cpu_threshold")
+
+        # Record memory status at start
+        memory_output = duthost.shell("show system-memory")["stdout"]
+        logging.info("Memory Status at start: %s", memory_output)
+
+        # Record Redis Memory at start
+        start_time_redis_memory = duthost.shell("redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\\1/'")["stdout"]
+        logging.info("Redis Memory: %s M", start_time_redis_memory)
+
+        # Record ipv4 route counts at start
+        start_time_ipv4_route_counts = duthost.shell("show ip route summary | grep Total | awk '{print $2}'")["stdout"]
+        logging.info("IPv4 routes at start: %s", start_time_ipv4_route_counts)
+
+        # Record ipv6 route counts at start
+        start_time_ipv6_route_counts = duthost.shell("show ipv6 route summary | grep Total | awk '{print $2}'")["stdout"]
+        logging.info("IPv6 routes at start %s", start_time_ipv6_route_counts)
+
+        # Make Sure Orch CPU < orch_cpu_threshold before starting test.
+        logging.info("Make Sure orchagent CPU utilization is less that %d before link flap", orch_cpu_threshold)
+        pytest_assert(wait_until(100, 2, check_orch_cpu_utilization, duthost, orch_cpu_threshold),
+                  "Orch CPU utilization {} > orch cpu threshold {} before link flap"
+                  .format(duthost.shell("show processes cpu | grep orchagent | awk '{print $9}'")["stdout"], orch_cpu_threshold))
+
+        # Flap all interfaces one by one on DUT
+        for iteration in range(3):
+            logging.info("%d Iteration flap all interfaces one by one on DUT", iteration + 1)
+            port_toggle(duthost, watch=True)
+
+        # Flap all interfaces one by one on Peer Device
+        for iteration in range(3):
+            logging.info("%d Iteration flap all interfaces one by one on Peer Device", iteration + 1)
+            candidates = build_test_candidates(duthost, fanouthosts, 'unknown')
+
+            if not candidates:
+                pytest.skip("Didn't find any port that is admin up and present in the connection graph")
+
+            for dut_port, fanout, fanout_port in candidates:
+                toggle_one_link(duthost, dut_port, fanout, fanout_port, watch=True)
+
+        # Wait 60 secs so that BGP routes are relearned
+        time.sleep(60)
```

Comment:
Can you change this to a wait_until so that we don't sleep extra time?

Reply:
Makes sense. Done